R/commonFunctionsAcceptSampling.R

```diff
@@ -0,0 +1,759 @@
+      x <- data.frame(X1 = seq(c.s[1], r.s[1], by=1),
+                      X.acc = c[2]-seq(c.s[1], r.s[1], by=1),
+                      X.rej = r[2]-1-seq(c.s[1], r.s[1], by=1))
+      p.acc_2 <- sum(apply(x, 1, FUN=prob.acc, n=n, p=pd, dist=dist, N=N))
+      p.acc <- p.acc + p.acc_2
+      acc_probs[k,] <- p.acc_2
+      p.rej_2 <- sum(apply(x, 1, FUN=prob.rej, n=n, p=pd, dist=dist, N=N))
+      p.rej <- p.rej + p.rej_2
+      rej_probs[k,] <- p.rej_2
+    }
+    else {
+      # More than two sampling stages.
+      c.s <- c+1 # Use to calculate limits
+      r.s <- r-1 # Use to calculate limits
+      expand.call <- "expand.grid(c.s[k-1]:r.s[k-1]"
+      for (i in 2:(k-1)) {
+        expand.call <- paste(expand.call, paste("c.s[k-",i,"]:r.s[k-",i,"]", sep=""), sep=",")
+      }
+      expand.call <- paste(expand.call,")", sep="")
+      x <- eval(parse(text=expand.call)[[1]])
+      x <- x[,(k-1):1] # Reverses the order of columns in dataframe x
+      names(x) <- paste("X", 1:(k-1), sep="")
+      
+      for (i in ncol(x):2) {
+        x[,i] <- x[,i] - x[,i-1]
+      }
+      x <- cbind(x, X.acc = c[k] - rowSums(x[,1:(k-1)]))
+      x <- cbind(x, X.rej = r[k]-1 - rowSums(x[,1:(k-1)]))
+      p.acc_k <- sum(apply(x, 1, FUN=prob.acc, n=n, p=pd, dist=dist, N=N))
+      p.acc <- p.acc + p.acc_k
+      acc_probs[k,] <- p.acc_k # Acceptance probability for the kth stage
+      p.rej_k <- sum(apply(x, 1, FUN=prob.rej, n=n, p=pd, dist=dist, N=N))
+      p.rej <- p.rej + p.rej_k
+      rej_probs[k,] <- p.rej_k # Rejection probability for the kth stage
+    }
+  }
+  return(list(acc_probs,rej_probs))
+}
+
+##------------------------------------------------------------------------
+##  Get stagewise acceptance and rejection probabilities for the plan.  --
+##------------------------------------------------------------------------
+#' @param pd {vector} Range of quality levels over which to evaluate the plan, each value indicating a proportion (0 to 1) of non-conforming items.
+#' @param n {vector} Sample size(s) for the plan.
+#' @param c {vector} Acceptance number(s) for the plan.
+#' @param r {vector} Rejection number(s) for the plan.
+#' @param dist {string} The type of distribution of non-conforming items in the sample. Possible values are "binom", "hypergeom", and "poisson".
+#' @param N {numeric} (optional) Total size of the lot.
+#' @returns list
+#' @seealso getStageProbabilityHelper()
+#' @examples
+#' getStageProbability(pd, n=c(10,20,30), c=c(2,5,8), r=c(4,7,9), dist="binom", N=1000)
+#' @notes
+#' The calculation of stagewise probability is done independently for every value in the vector 'pd', which has a range of quality levels.
+##------------------------------------------------------------------------
+getStageProbability <- function(pd, n, c, r, dist, N=10000) {
+  stage_probs <- sapply(pd, FUN=getStageProbabilityHelper, n=n, c=c, r=r, dist=dist, N=N)
```

Comment:
could you use `vapply` here? or altnernatively `lapply`, since you're creating your own output structure anyway the simplification step of `sapply` is kinda unnecessary.

Reply:
See message.

Reply:
@vandenman I did change sapply to vapply, but I noticed that the FUN.VALUE check within vapply isn't very strict. The variable it gets is a list of two matrices, and as long as I tell it to expect a list with 2 elements, it works fine. So list(1,2) works the same as list(matrix,matrix).